app/dts/bindings/behaviors/zmk,behavior-hold-tap.yaml

```diff
@@ -30,3 +30,7 @@ properties:
       - "tap-preferred"
   retro-tap:
     type: boolean
+  hold-trigger-key-positions:
```

Comment:
Is there a way I can create a compile-time constraints?
1. if `flavor = tap_positionally_preferred`, then `hold-trigger-key-positions` is included
2. if `hold-trigger-key-positions` is included, then `flavor = tap_positionally_preferred`

Reply:
You can but I don't recommend it.
I suggest adding something like this inside `KP_INST` for case 1:
```C
BUILD_ASSERT(!(DT_ENUM_IDX(DT_DRV_INST(n), flavor) == 3
                 && DT_INST_PROP_LEN(n, hold_trigger_key_positions) > 0),
             "flavor is 'tap-positionally-preferred' but 'hold-trigger-key-positions' is empty")
```
And something similar for case 2.

Reply:
why do you recommend against? given that this configuration option should be used if and only if the flavor is tap-positionally-preferred, it seems like it'd be good to have compile-time enforcement of that assumption?

Reply:
I think `BUILD_ASSERT` is a the best way to do that.
There are a few ways to get these checks done in the devicetree but they are much more complicated and I don't recommend them as they may break backward compatibility.